docs/source/howto/exploring.rst

```diff
@@ -0,0 +1,137 @@
+
+As an example:
+
+.. code-block:: python
+
+    node.get_incoming(node_class=Data, link_type=LinkType.INPUT_CALC, link_label_filter='output%node_').all_nodes()
+
+will return only neighboring data nodes that are linked to the ``node`` with a link of type ``LinkType.INPUT_CALC`` and where the link label matches the regex ``'output%node_'``.
+The ``%`` character matches one or more characters and the ``_`` character matches exactly one character.
+
+
+.. _how-to:exploring:inputs-outputs:
+
+Inputs and outputs
+==================
+
+The :meth:`~aiida.orm.nodes.node.Node.get_incoming` and :meth:`~aiida.orm.nodes.node.Node.get_outgoing` methods, described in the :ref:`previous section <how-to:exploring:incoming-outgoing>`, can be used to access all neighbors from a certain node and provide advanced filtering options.
+However, often one doesn't need this expressivity and simply wants to retrieve all neighboring nodes with a syntax that is as succint as possible.
+A prime example is to retrieve the *inputs* or *outputs* of :ref:`a process <topics:processes:concepts>`.
+Instead of using :meth:`~aiida.orm.nodes.node.Node.get_incoming` and :meth:`~aiida.orm.nodes.node.Node.get_outgoing`, to get the inputs and outputs of a ``process_node`` one can do:
+
+.. code-block:: python
+
+    inputs = process_node.inputs
+    outputs = process_node.outputs
+
+These properties do not return the actual inputs and outputs directly, but instead return an instance of :class:`~aiida.orm.utils.managers.NodeLinksManager`
+The reason is because through the manager, the inputs or outputs are accessible through their link label and it is tab-completed.
+For example, if the ``process_node`` has an output with the label ``result``, it can be retrieved as:
+
+.. code-block:: python
+
+    process_node.outputs.result
+
+The inputs or outputs can also be accessed through key dereferencing:
+
+.. code-block:: python
+
+    process_node.outputs['result']
+
+If there is no neighboring output with the given link label, a :class:`~aiida.common.exceptions.NotExistentAttributeError` or :class:`~aiida.common.exceptions.NotExistentKeyError` will be raised, respectively.
+
+.. note::
+
+    The ``inputs`` and ``outputs`` properties are only defined for :class:`~aiida.orm.nodes.process.process.ProcessNode`'s.
+    This means that you cannot *chain* these calls, because an input or output of a process node is guaranteed to be a :class:`~aiida.orm.nodes.data.Data` node, which does not have inputs or outputs.
+
+
+.. _how-to:exploring:creator-caller-called:
+
+Creator, caller and called
+==========================
+
+Similar to the ``inputs`` and ``outputs`` properties of process nodes, there are some more properties that make exploring the provenance graph easier:
+
+    * :meth:`~aiida.orm.nodes.process.process.ProcessNode.called`: defined for :class:`~aiida.orm.nodes.process.process.ProcessNode`'s and returns the list of process nodes called by this node.
```

Comment:
Not relevant for this PR, but I realise now that we could have defined this only for WorkflowNodes, actually, since the list will always be empty for CalculationNodes

Reply:
Yeah, and I think the `ProcessNode` should also have a `inputs` and `outputs` implementation. Now I cannot reference it